docker/spec/fixtures/docker/dockerfiles/multi_stage_global_arg

```diff
@@ -0,0 +1,26 @@
+ARG REGISTRY=registry-host.io
+FROM ${REGISTRY}/ubuntu:17.04 AS BUILD
+
+### SYSTEM DEPENDENCIES
+
+RUN apt-get update \
+    && apt-get upgrade -y \
+    && apt-get install -y --no-install-recommends \
+      build-essential \
+      dirmngr \
+      git \
+
+
+### RUBY
+
+# Install Ruby 2.4
+RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys C3173AA6 \
+    && echo "deb http://ppa.launchpad.net/brightbox/ruby-ng/ubuntu zesty main" > /etc/apt/sources.list.d/brightbox.list \
+    && apt-get update
+RUN apt-get install -y ruby2.4 ruby2.4-dev
+
+# This should have no impact on global ARG instructions used in FROM instructions
+ARG REGISTRY=docker.io
+FROM ${REGISTRY}/ubuntu:17.04
```

Comment:
Would docker use `docker.io` here? Think this will cause pain down the line if we diverge from Docker's behaviour. Could we rework this to use the last found ARG when matching a used arg in `FROM`?

Reply:
Yes, `docker.io` is the default behavior in docker. On the other hand this is just a test example.

Reply:
Ok, I understand your comment the wrong way. No, docker would not use docker.io here, as there is a difference between global (before the first `FROM`) and local `ARG` instructions. The test is correct.